### PR TITLE
verification-add-attribute

### PR DESCRIPTION
**Pull request by marjan-ara**

### CODE REVIEW

File: src/modules/Verifications/components/Atribute/index.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        <SearchDropdown
+          value={selectedOperator}
+          options={operators}
+          isSearchable
+          onChange={onSelectOperator}
+          containerClassName="flex-1 w-full"
+          errors={errors && errors['operator']?.message ? [errors['operator'].message] : undefined}
+        />
+        <Input
+          value={attributeValue}
```
Comment (satanmourner) [inline src/modules/Verifications/components/Atribute/index.tsx:48]: why not use `register`?
P.S i want to add for all component controller to be handle by controller and not in uncontrolled way

File: src/modules/Verifications/components/Atribute/useAttribute.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    { value: 'EQUAL', label: 'Is equal to' },
+    { value: 'NOT', label: 'Is not' },
+    { value: 'BIGGER', label: 'Is greater than' },
+    { value: 'SMALLER', label: 'Is less than' },
+  ];
+
+  const [selected, setSelected] = useState<OptionType>({ label: attribute?.name || '', value: attribute?.id || '' });
+  const [selectedOperator, setSelectedOperator] = useState<OptionType>({
+    value: attribute?.operator || '',
+    label: attribute ? operators.find(item => item.value === attribute.operator)?.label || '' : '',
```
Comment (satanmourner) [inline src/modules/Verifications/components/Atribute/useAttribute.tsx:19]: more readable in this way:
`operators?.find(item => item.value === attribute.operator)?.label || ''`

File: src/pages/verifications/createUpdateVerification/useCreateUpdateVerification.tsx [hunk 3]
```diff
@@ -113,6 +158,34 @@ export const useCreateUpdateVerification = () => {
   const name = watch('name');
   const description = watch('description');
 
+  const onChangeAttribute = (index: number, attribute?: OptionType, operator?: OptionType, value?: string) => {
+    const newAttributes = getValues().attributes;
+    if (attribute !== undefined) {
```
Comment (satanmourner) [inline src/pages/verifications/createUpdateVerification/useCreateUpdateVerification.tsx:163]: specifically `undefined` ? if attribute is null, what happens?

**Review by satanmourner — CHANGES_REQUESTED** [review]

File: src/modules/Verifications/components/Atribute/useAttribute.tsx (hunk 2 above)
Comment (marjan-ara) [reply src/modules/Verifications/components/Atribute/useAttribute.tsx:19]: I cannot remove the condition of checking attribute. it is practical to distinguish if user is creating a new verification or updating an old one. It also causes type script error
For example if user is creating a new verification, attribute is empty, so the find operation shouldn't execute

File: src/pages/verifications/createUpdateVerification/useCreateUpdateVerification.tsx (hunk 3 above)
Comment (marjan-ara) [reply src/pages/verifications/createUpdateVerification/useCreateUpdateVerification.tsx:163]: This function doesn't accept null for attribute. it is either undefined or has value

Comment (satanmourner) [reply src/pages/verifications/createUpdateVerification/useCreateUpdateVerification.tsx:163]: i mean you can easilyc check with `(attribute)` as truthy value

File: src/modules/Verifications/components/Atribute/index.tsx (hunk 1 above)
Comment (marjan-ara) [reply src/modules/Verifications/components/Atribute/index.tsx:48]: register just setValues for property,
Here I wanna control an attribute values comparing to other attributes of verification, for example, I should check that not a duplicate attribute is entered, so error handling and setting attribute value is done in the parent component,

**Review by satanmourner — APPROVED** [review]

**Pull request merged**